### PR TITLE
Storing which model has been used to embed source

### PR DESCRIPTION
**Pull request by Til7701**
I decided not to store the model used for embedding in the document metadata. Instead it is stored in our tables for sources. This way a source can be embedded with multiple models at once. For each model, the vector store will use a different table. This way old data can be dropped easily and sources can be split and embedded based on the original source if the model changes. This also minimizes extra metadata in the document with lots of duplicates.

### CODE REVIEW

File: src/main/java/de/uol/pgdoener/civicsage/config/VectorStoreTableNameEnvironmentPostProcessor.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
+        if (environment.containsProperty(TABLE_NAME_PROPERTY_NAME)) {
+            return; // Property has been overwritten externally
+        }
+
+        Object o = environment.getProperty(MODEL_PROPERTY_NAME);
+        if (o instanceof String model) {
+            // build table name from model name
+            String tableName = "vector_store_" + model.replace("/", "_");
```
Comment (JayPi4c) [inline src/main/java/de/uol/pgdoener/civicsage/config/VectorStoreTableNameEnvironmentPostProcessor.java:35]: I'm currently playing around a little with different models (on another node: Spring dislikes most models I tested because due to `Pooling type 'none' is not OAI compatible. Please use a different pooling type`. I'm yet to figure out how to deal with this.)

Back to this topic: I came upon the following model name: `Qwen/Qwen3-Embedding-0.6B-GGUF` 

With this model the app can't start because of `Caused by: org.postgresql.util.PSQLException: ERROR: trailing junk after numeric literal at or near ".6B_GGUF"`

File: src/main/java/de/uol/pgdoener/civicsage/source/FileSource.java [hunk 2]
```diff
@@ -24,4 +26,7 @@ public class FileSource {
     @Column(unique = true)
     private String hash;
 
+    @ElementCollection
+    private List<String> models;
```
Comment (JayPi4c) [inline src/main/java/de/uol/pgdoener/civicsage/source/FileSource.java:30]: Currently, I get a 409 Conflict, when trying to embed the same website twice. So currently it is impossible to embed a website with a different model, rendering the `List` useless. But it's just to be ready for future changes, right?

**Review by JayPi4c — CHANGES_REQUESTED** [review]
As far as I can tell, multiple embeddings of the same resource is still not possible, due to duplicate prevention. But this is probably not scope of this PR.

Yet the name conversion for the db name has to be fixed, before this can be merged.

File: src/main/java/de/uol/pgdoener/civicsage/config/VectorStoreTableNameEnvironmentPostProcessor.java (hunk 1 above)
Comment (Til7701) [reply src/main/java/de/uol/pgdoener/civicsage/config/VectorStoreTableNameEnvironmentPostProcessor.java]: This should be fixed now.

File: src/main/java/de/uol/pgdoener/civicsage/source/FileSource.java (hunk 2 above)
Comment (Til7701) [reply src/main/java/de/uol/pgdoener/civicsage/source/FileSource.java:30]: I actually forgot that. But this PR is nicely contained right now. Adding this would blow it up. I'll create a new issue for that.

Comment (Til7701) [reply src/main/java/de/uol/pgdoener/civicsage/source/FileSource.java:30]: #62
